website/project/spam/__init__.py

```diff
@@ -0,0 +1,109 @@
+from framework.logging import logger
+from framework.celery_tasks import app as celery_app
+from framework.mongo import get_cache_key as get_request
+
+from website import settings
+from website.util import akismet, get_headers_from_request
+from website.project.licenses import serialize_node_license_record
+
+NODE_SPAM_FIELDS = set((
+    'title',
+    'description',
+    'wiki_pages_current',
+    'node_license'
+))
+
+def _get_client():
+    return akismet.AkismetClient(
+        apikey=settings.AKISMET_APIKEY,
+        website=settings.DOMAIN,
+        verify=True
+    )
+
+def _get_content(node):
+    from website.addons.wiki.model import NodeWikiPage
+    content = []
+    for field in NODE_SPAM_FIELDS:
+        if 'wiki' in field:
+            content.append('\n'.join([
+                NodeWikiPage.load(x).raw_text(node)
+                for x in node.wiki_pages_current.values()
+            ]))
+            continue
+        if field == 'node_license':
+            content.append(serialize_node_license_record(node.license).get('text', ''))
+            continue
+        content.append((getattr(node, field, None) or '').encode('utf-8'))
+    return '\n'.join(content)
+
+
+@celery_app.task(ignore_result=True)
+def confirm_spam(node_id):
+    from website.models import Node
+    node = Node.load(node_id)
+    request_headers = get_headers_from_request(
+        get_request()
```

Comment:
The request parameters must be passed from to the worker job as inputs, this request will no longer be the the http request made by the client.